operatorcourier/cli.py

```diff
@@ -45,11 +48,18 @@ def parse(self):
         parser.add_argument('-v', '--version',
                             help='Show the current version of operator-courier',
                             action='version', version=__version__)
+        parser.add_argument('--log-level', dest='log_level',
+                            help="Set logging level (default: WARNING)",
+                            choices=['DEBUG', 'INFO', 'WARNING', 'ERROR',
+                                     'CRITICAL'],
+                            default='WARNING', type=str.upper)
 
         args = parser.parse_args(sys.argv[1:2])
```

Comment:
This PR is not working due this line. Because PR with argparse subparsers was rejected, I'm waiting for suggestion how to update this.

Reply:
@MartinBasti 

Two comments here:

1. I'm still not sure I see why not using subparsers is preventing a top level flag like this. Why can't we just explicitly parse this command in this method? We aren't throwing anything away.

2. The PR that included a move to subparsers wasn't rejected because it used subparsers. It was rejected because the PR was attempting to address two unrelated things (adding subparsers and attempting to resolve this issue: https://github.com/operator-framework/operator-courier/issues/77). I just stated that moving to subparsers didn't fix that issue, and either the PR should have been updated to fix that issue and explain why adding subparser use was needed to resolve it OR stop referencing that issue and instead just add subparsers and in a detailed way explain why that was useful. Instead of following up on my point, the PR was just closed.

I am not against using subparsers, I just believe that git history should be concise and clean and that every PR should have a singular stated purpose. If you strongly believe that the best way to implement this feature is to include the use of subparsers then feel free to either reopen the other PR with what I was requesting, or feel free to include that change in this PR if you feel it is necessary for this change. But when you do, please clearly state why that is the case.

Reply:
1. 
* main parser uses only first argument `sys.argv[1:2]`
* command parsers uses second+ argument for parsing `sys.argv[2:]`
* when I use `--log-level` on first position I got error that subcommand is required
* when I use subcommand on first position, log-level is just ignored
* because log-level is optional, I cannot hardcode `sys.argv[3:]`
* I'm not gonna copy that option to each command parser
* subparsers handles this even without thinking about that some `sys.argv[]` exists

2. I will reopen it then